File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/ReviewDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for individual course review.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReviewDTO {
    private Long id;
    private String content;
    private String contentHtml;
    private Float rating;
    private String created;
    private String createdFormatted; // e.g., "4 days ago"
    private ReviewUserDTO user;
    private ReviewResponseDTO response; // Instructor response, if any
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/ReviewDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for individual course review.
 */
data class ReviewDTO(
    val id: Long? = null,
    val content: String? = null,
    val contentHtml: String? = null,
    val rating: Float? = null,
    val created: String? = null,
    // e.g., "4 days ago"
    val createdFormatted: String? = null,
    val user: ReviewUserDTO? = null,
    // Instructor response, if any
    val response: ReviewResponseDTO? = null,
)
